SurfaceTopography/IO/NC.py

```diff
@@ -230,33 +305,108 @@ def write_nc(topography, filename, format='NETCDF3_64BIT_OFFSET'):
             topography.communicator.rank > 1:
         return
     with Dataset(filename, 'w', **kwargs) as nc:
-        nx, ny = topography.nb_grid_pts
-        sx, sy = topography.physical_sizes
+        # Serialize info dictionary as JSON and write to NetCDF file
+        info = topography.info.copy()
+        try:
+            # We store the unit information separately
+            del info['unit']
+        except KeyError:
+            pass
+        if info != {}:
+            nc.json = json.dumps(info, ensure_ascii=True, cls=NumpyEncoder)
+
+        # Create dimensions and heights variable
+        try:
+            nx, ny = topography.nb_grid_pts
+        except ValueError:
+            nx, = topography.nb_grid_pts
 
         nc.createDimension('x', nx)
-        nc.createDimension('y', ny)
+        if topography.dim > 1:
+            nc.createDimension('y', ny)
+            heights_var = nc.createVariable('heights', 'f8', ('x', 'y'))
+        else:
+            heights_var = nc.createVariable('heights', 'f8', ('x',))
 
-        x_var = nc.createVariable('x', 'f8', ('x',))
-        y_var = nc.createVariable('y', 'f8', ('y',))
-        heights_var = nc.createVariable('heights', 'f8', ('x', 'y',))
+        # Create variables for x- and y-positions, but only if physical_sizes
+        # exist. (physical_sizes should always exist, but who knows...)
+        if topography.physical_sizes is not None:
+            try:
+                sx, sy = topography.physical_sizes
+            except ValueError:
+                sx, = topography.physical_sizes
 
-        x_var.length = sx
-        x_var.periodic = 1 if topography.is_periodic else 0
-        if 'unit' in topography.info:
-            # scipy.io.netcdf_file does not support UTF-8
-            x_var.length_unit = no_uft8_height_unit(topography.info['unit'])
-        x_var[...] = (np.arange(nx) + 0.5) * sx / nx
-        y_var.length = sy
-        y_var.periodic = 1 if topography.is_periodic else 0
-        if 'unit' in topography.info:
-            # scipy.io.netcdf_file does not support UTF-8
-            y_var.length_unit = no_uft8_height_unit(topography.info['unit'])
-        y_var[...] = (np.arange(ny) + 0.5) * sy / ny
+            x_var = nc.createVariable('x', 'f8', ('x',))
+            x_var.length = sx
+
+            x_var.periodic = 1 if topography.is_periodic else 0
+            if 'unit' in topography.info:
+                # scipy.io.netcdf_file does not support UTF-8
+                x_var.unit = mangle_length_unit_ascii(topography.info['unit'])
+            x_var[...] = (np.arange(nx) + 0.5) * sx / nx
```

Comment:
Should we be efficient and only store the physical sizes instead of the full array ?

Reply:
This is simply for convenience. If you open the NetCDF file in Python or MATLAB, you can just play heights vs x and that shows the topography. This works for nonuniform and uniform cases.

Reply:
Ok makes sense.